mesonbuild/mlog.py

```diff
@@ -421,17 +421,22 @@ def start_pager() -> None:
                 path = Path(git).parents[1] / 'usr' / 'bin'
                 less = shutil.which('less', path=str(path))
         if less:
-            # "R" : support color
-            # "X" : do not clear the screen when leaving the pager
-            # "F" : skip the pager if content fit into the screen
-            pager_cmd = [less, '-RXF']
+            pager_cmd = [less]
     if not pager_cmd:
         return
     global log_pager # pylint: disable=global-statement
     assert log_pager is None
     try:
+        # Set 'LESS' environment variable, rather than arguments in
+        # pager_cmd, to also support the case where the user has 'PAGER'
+        # set to 'less'. Arguments set are:
+        # "R" : support color
+        # "X" : do not clear the screen when leaving the pager
+        # "F" : skip the pager if content fits into the screen
+        env = os.environ.copy()
+        env['LESS'] = "RXF"
```

Comment:
Probably we should only do this if $LESS is not already set.

Reply:
Indeed, and git does that too.

Reply:
Also while are it, git seems to set PV=-c

Reply:
Yeah good point, I've made that change. I think git just completely overwrites the environment's behaviour, but I guess if `LESS` is set the user really expects those options to be used.

Reply:
Woops, didn't see your extra comments, and I missed git's `getenv` check. Fixed.